app/models/scheme.rb

```diff
@@ -254,4 +254,14 @@ def reactivating_soon?
   def deactivated?
     status == :deactivated
   end
+
+  def has_confirmed_locations?
+    confirmed_locations_count.positive?
+  end
+
+private
+
+  def confirmed_locations_count
+    locations.confirmed.size
+  end
```

Comment:
if this one line method is only being called in one other one line method a few lines above I feel they could just be folded into one.
Personal preference is around 3 or 4 chained method calls to put it on separate lines
```
locations
  .confirmed
  .count
  .positive?
```

Reply:
actually if that proc above is the only place we're using this .confirmed then I feel like this would only be more readable and probably equally efficient if we just used active record directly?
`locations.count(confirmed: true).positive?

Reply:
again for readability I've done this separately as in CLDC-2140 I need unconfirmed_locations_count as well and pass them into a string so will keep this too

Reply:
You don't need either of these methods. You can just do:

```ruby
def status_at(date)
  return :incomplete unless confirmed && locations.confirmed.any?
  # ... 
```

Reply:
💣 💣 💣 💣 💣 💣

Reply:
that is much tidier @jamesrose I agree, I still stand by the point about readability in connection to CLDC-2140 however - specifically https://github.com/communitiesuk/submit-social-housing-lettings-and-sales-data/pull/1455/files#diff-0cd65118d88b34f32b919f37a20a1024d03620f8a892f3f020fe70b0193e5a61R169 where I do think this having confirmed and unconfirmed counts is the best way of working (unless there's a VERY brief active record query I could use?), and given that I think it makes sense to use those methods here too